### ISSUE TITLE
Consider component-ify-ing the badge

### ISSUE THREAD

**Issue by ang-zeyu** [MarkBind/markbind#2090]
### Please confirm that you have searched existing issues in the repo

Yes, I have searched the existing issues

### Is your feature request related to a problem? Please describe.

We still use raw bootstrap classes for the badge component.

### Describe the solution you'd like

We can consider component-ify-ing this ([example](https://bootstrap-vue.org/docs/components/badge#contextual-variations)) to make for usability consistency.

The priority is likely low as the syntax is fairly short and easy to use already.

**Comment by ang-zeyu** [comment]
@damithc having your user input here especially would be great, not sure if this is something actually useful.

**Comment by damithc** [comment]
> @damithc having your user input here especially would be great, not sure if this is something actually useful.

Good idea @ang-zeyu We can certainly consider it if we can come up with a syntax that is simpler than the Bootstrap syntax. Badges are certainly useful to make a content-heavy page easier on the eyes, and hence, having an optimized badges syntax is in our interest.

**Comment by yucheng11122017** [comment]
VuePress also has a [simpler syntax](https://vuepress.vuejs.org/guide/using-vue.html#built-in-components) for their badges. Although I think that the example given in the issue would be better since they only allow strings within the badge instead of other components.
